### PR TITLE
packetry: init at 0.1.0

### PR DESCRIPTION
**Pull request by carlossless**
## Description of changes

A fast, intuitive USB 2.0 protocol analysis application for use with Cynthion. https://github.com/greatscottgadgets/packetry

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pa/packetry/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    cairo
+    pango
+    gtk4
+    graphene
+  ];
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A fast, intuitive USB 2.0 protocol analysis application for use with Cynthion.";
```
Comment (afh) [inline pkgs/by-name/pa/packetry/package.nix:43]: According to the [recommendations for meta attributes](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) the description should not start with an article and provide factual information while avoid subjective language:
```suggestion
    description = "USB 2.0 protocol analysis application for use with Cynthion.";
```

File: pkgs/by-name/pa/packetry/package.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  graphene,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "packetry";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "greatscottgadgets";
+    repo = pname;
```
Comment (afh) [inline pkgs/by-name/pa/packetry/package.nix:21]: Please replace uses of `pname` with the actual package name, for further details please see #277994.
```suggestion
    repo = "packetry";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @carlossless, please find below two minor suggestions for improvement.

**Comment by carlossless** [comment]
Thanks for the review & suggestions @afh!

**Comment by afh** [comment]
Result of `nixpkgs-review pr 332070` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>packetry</li>
  </ul>
</details>

**Review by afh — APPROVED** [review]
Thanks for accepting the review suggestions and making the changes. This LGTM 🙂 👍

**Comment by carlossless** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/349828

**Pull request closed without merging**